src/main/resources/migrations/common/V2024.04.22__baseline_tables.sql

```diff
@@ -0,0 +1,328 @@
+    telephone            varchar(20),
+    position             varchar(100),
+    enabled              boolean NOT NULL,
+    notes                varchar(200),
+    created_by           varchar(100) NOT NULL,
+    created_time         timestamp NOT NULL,
+    amended_by           varchar(100),
+    amended_time         timestamp
+);
+
+CREATE INDEX idx_probation_contact_team_id ON probation_team_contact(probation_team_id);
+CREATE INDEX idx_probation_contact_name ON probation_team_contact(name);
+CREATE INDEX idx_probation_contact_email ON probation_team_contact(email);
+
+---------------------------------------------------------------------------------------
+-- Reference table for prisons - populate from register
+---------------------------------------------------------------------------------------
+
+CREATE TABLE prison
+(
+    prison_id    bigserial   NOT NULL CONSTRAINT prison_id_pk PRIMARY KEY,
+    code         varchar(10) NOT NULL UNIQUE,
+    name         varchar(60) NOT NULL,
+    description  varchar(100) NOT NULL,
+    enabled      boolean NOT NULL,
+    notes        varchar(200),
+    created_by   varchar(100) NOT NULL,
+    created_time timestamp NOT NULL,
+    amended_by   varchar(100),
+    amended_time timestamp
+);
+
+CREATE INDEX idx_prison_code ON prison(code);
+
+---------------------------------------------------------------------------------------
+-- Contacts for this prison
+---------------------------------------------------------------------------------------
+
+CREATE TABLE prison_contact
+(
+    prison_contact_id  bigserial NOT NULL CONSTRAINT prison_contact_id_pk PRIMARY KEY,
+    prison_id bigint NOT NULL REFERENCES prison(prison_id),
+    name varchar(100) NOT NULL,
+    email varchar(100),
+    telephone varchar(20),
+    position varchar(100),
+    enabled boolean NOT NULL,
+    notes varchar(200),
+    created_by   varchar(100) NOT NULL,
+    created_time timestamp NOT NULL,
+    amended_by   varchar(100),
+    amended_time timestamp
+);
+
+CREATE INDEX idx_prison_prison_id ON prison_contact(prison_id);
+CREATE INDEX idx_prison_contact_name ON prison_contact(name);
+CREATE INDEX idx_prison_contact_email ON prison_contact(email);
+
+---------------------------------------------------------------------------------------
+-- This is the main table for a video booking.
+-- It has a type - PROBATION or COURT booking (can extend to others)
+-- court_id / hearing_type / probation_team_id / probation_meeting_type  
+--     - are nullable FK references
+----------------------------------------------------------------------------------------
+
+CREATE TABLE video_booking
+(
+    video_booking_id       bigserial NOT NULL CONSTRAINT video_booking_pk PRIMARY KEY,
+    booking_type           varchar(40) NOT NULL,  -- COURT, PROBATION
+    status_code            varchar(20) NOT NULL,  -- ACTIVE, COMPLETED, EXPIRED, CANCELLED
+    court_id               bigint REFERENCES court(court_id), -- Nullable
```

Comment:
I am still not 100% sure on this. Relies on code enforcing the rules.

I wonder, is it possible to add constraints at the DB level e.g. one or the other must have a value?

Reply:
Don't know a way to constrain either/or - unless we have a pre-insert trigger and a procedure/function to check?